app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -0,0 +1,80 @@
+module ManageIQ::Providers
+  module Hawkular
+    class MiddlewareManager::RefreshParser
+      def self.ems_inv_to_hashes(ems, options = nil)
+        new(ems, options).ems_inv_to_hashes
+      end
+
+      def initialize(ems, _options = nil)
+        @ems = ems
+        @eaps = []
+        @data = {}
+        @data_index = {}
+      end
+
+      def ems_inv_to_hashes
+        @data[:middleware_servers] = fetch_middleware_servers
+        fetch_deployments
+        @data
+      end
+
+      def fetch_middleware_servers
```

Comment:
The pattern is to use the prefix `get_` instead of `fetch_` (yes, I know rubocop complains, but this is one of those cases that we ignore).